### PR TITLE
Add a section describing a Time Server compromise

### PR DESCRIPTION
**Pull request by mnm678**
Solves #45

This pr adds a description of the consequences of a Time Server compromise. I didn't add much detail about how to recover from this attack, as I felt it is well described on the Key Management page, but I can add a section if others disagree.

### CODE REVIEW

File: repositories.md [hunk 1]
```diff
@@ -72,6 +72,14 @@ In order to prevent a new Time Server from accidentally causing a rollback warni
...
 
+#### Time Server key compromise
+
+In the event of a Time Server key compromise, an attacker would be able to return a time attestation that contains an arbitrary time. The attacker could then either:
+* Make all metadata appear expired. If the Time Server returns a time far in the future, all Uptane metadata will appear expired to the vehicle. The vehicle would be unable to verify the metadata, thus creating a denial of service.
+* Make expired metadata appear current. If the Time Server returns a time in the past, Uptane metadata that was valid at that point in the past will appear valid to the vehicle, thus allowing for a freeze attack. This cannot be used for a rollback attack as the ECU will not accept a time earlier than the time of their previous update.
+
+All of these attacks can be mitigated by rotating the Time Server key in root metadata, as described in [Managing signing keys and metadata expiration](https://uptane.github.io/deployment-considerations/key_management.html).
```
Comment (pattivacek) [inline repositories.md:81]:
```suggestion
All of these attacks can be mitigated by rotating the Time Server key in Root metadata, as described in [Managing signing keys and metadata expiration](https://uptane.github.io/deployment-considerations/key_management.html).
```

**Review by pattivacek — COMMENTED** [review]
LGTM minus one capitalization!

File: repositories.md [hunk 2]
```diff
@@ -72,6 +72,14 @@ In order to prevent a new Time Server from accidentally causing a rollback warni
 
 As partial verification Secondaries only check the Targets metadata from the Director repository, the Time Server key on these ECUs will be checked when verifying the Targets metadata. To do this, check the Time Server key after verifying the most recent Targets metadata file. If the Time Server key is listed in the Targets metadata and has been rotated, reset the clock used to determine the expiration of metadata to a minimal value as described above.
 
+#### Time Server key compromise
+
+In the event of a Time Server key compromise, an attacker would be able to return a time attestation that contains an arbitrary time. The attacker could then either:
+* Make all metadata appear expired. If the Time Server returns a time far in the future, all Uptane metadata will appear expired to the vehicle. The vehicle would be unable to verify the metadata, thus creating a denial of service.
```
Comment (jhdalek55) [inline repositories.md:78]: In the event of a Time Server key compromise, an attacker would be able to return a time attestation that contains an arbitrary time. The attacker could then either:
* Make all metadata appear to be expired. If the Time Server returns a time far in the future, the vehicle would interpret it as expired Uptane metadata.  The vehicle would be unable to verify the metadata, thus creating a denial of service.

File: repositories.md [hunk 3]
```diff
@@ -72,6 +72,14 @@ In order to prevent a new Time Server from accidentally causing a rollback warni
...
+* Make expired metadata appear current. If the Time Server returns a time in the past, Uptane metadata that was valid at that point in the past will appear valid to the vehicle, thus allowing for a freeze attack. This cannot be used for a rollback attack as the ECU will not accept a time earlier than the time of their previous update.
```
Comment (jhdalek55) [inline repositories.md:79]: Make expired metadata appear to be current. If the Time Server returns a time in the past, Uptane metadata that was valid at that point in the past will seem current to the vehicle, thus allowing for a freeze attack. This cannot be used for a rollback attack as the ECU will not accept a time earlier than the time of their previous update.

**Review by jhdalek55 — COMMENTED** [review]
Looks good to me on the whole. Made a few sentence level suggestions.

**Comment by mnm678** [comment]
Thanks @pattivacek and @jhdalek55! I updated the pr with your suggestions.

**Comment by jhdalek55** [comment]
@allencain @iramcdonald @trishankatdatadog Is this sufficient?

**Comment by iramcdonald** [comment]
Hi,

LGTM

I just carefully read the whole thread in the issue.  I like the level of
detail and agree with
Marina that we should put recovery text inline in this update.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Comment by jhdalek55** [comment]
Any objections if I merge this? It has been reviewed by three of us.

**Review by pattivacek — APPROVED** [review]

**Comment by jhdalek55** [comment]
Closes Issue #45

**Pull request merged**